### PR TITLE
validate saas files are reusing environment parameters

### PR DESCRIPTION
**Pull request by maorfr**
we can force reusing parameters defined on the environment instead of redefining them in saas files.
this will help us rebuild complete environments faster and update full environments to point to new endpoints, such as dependencies.

### CODE REVIEW

**Comment by apahim** [comment]
on it

**Comment by maorfr** [comment]
i'm aware we can re-use a lot of the code, but i'd rather refactor in a future iteration

File: tools/qontract_cli.py [hunk 1]
```diff
@@ -517,6 +517,18 @@ def saas_dev(ctx, app_name=None, saas_file_name=None, env_name=None):
...
+                for replace_key, replace_value in parameters.items():
+                    if not isinstance(replace_value, str):
+                        continue
+                    replace_pattern = '${' + replace_key + '}'
+                    for k, v in parameters.items():
+                        if not isinstance(v, str):
+                            continue
+                        if replace_pattern in v:
+                            parameters[k] = \
+                                v.replace(replace_pattern, replace_value)
```
Comment (apahim) [inline tools/qontract_cli.py:530]: Not sure I understand what you're trying to achieve here. Mind explaining?

Comment (maorfr) [reply tools/qontract_cli.py:530]: if someone uses the following parameter: `AUTHORIZE_URL: ${OCM_BASE_URL}/auth`
we replace the `OCM_BASE_URL` with the actual value coming from the environment.
this is actually the parameter unpacking.

Comment (apahim) [reply tools/qontract_cli.py:530]: I see. `validate saas files are reusing environment parameters` or the commit messges do not tell the full story here :)
tks.

**Review by apahim — APPROVED** [review]
You know me so well... I was going to comment on the code duplication :)
Please don't loose track of it.

**Pull request merged**